tests/unittests/MLTest.cpp

```diff
@@ -205,6 +207,79 @@ TEST_P(MLTest, learnXor) {
   }
 }
 
+/// Learn the logarithmic function
+TEST_P(MLTest, learnLog) {
+  unsigned numInputs = 500;
+  unsigned batchSize = 10;
+  EE_.getConfig().learningRate = 0.05;
+  EE_.getConfig().batchSize = batchSize;
+
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("main");
+  F->setName("learnLog");
+
+  auto *A =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "A", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+  auto *Ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "Ex", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+
+  Node *O = F->createFullyConnected("fc1", A, 5);
+  O = F->createTanh("tanh1", O);
+  O = F->createFullyConnected("fc2", O, 6);
+  O = F->createTanh("tanh2", O);
+  O = F->createFullyConnected("fc3", O, 1);
+  O = F->createTanh("tanh3", O);
+  O = F->createRegression("reg", O, Ex);
+  auto *result = F->createSave("ret", O);
+
+  //Set the training set 
+  Tensor trainingSet(ElemKind::FloatTy, {numInputs, 1});
+  Tensor trainingLabels(ElemKind::FloatTy, {numInputs, 1});
+
+  auto TS = trainingSet.getHandle<>();
+  auto TL = trainingLabels.getHandle<>();
+
+  float LO = 0.75;
+  float HI = 1.5;
+  srand (static_cast <unsigned> (time(0)));
+  for (size_t i = 0; i < numInputs; i ++) {
+    float a = LO + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(HI-LO)));
+    TS.at({i,0}) = a;
+    TL.at({i,0}) = std::log(a);
+  }
+
+  Function *TF = glow::differentiate(F, EE_.getConfig());
+  EE_.compile(CompilationMode::Train, TF);
+
+  // Train the network:
+  EE_.runBatch(2500, {A, Ex}, {&trainingSet, &trainingLabels});
+
+  EE_.compile(CompilationMode::Infer, F);
+
+  //Set the testing set
+
+  Tensor testSet(ElemKind::FloatTy, {batchSize, 1});
+
+  auto TES = testSet.getHandle<>();
+
+  for (size_t i = 0; i < batchSize; i ++) {
+    float a = LO + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(HI-LO)));
+    TES.at({i,0}) = a;
+  }
+
+  EE_.run({A}, {&testSet});
+  auto resH = result->getVariable()->getPayload().getHandle<>();
+
+  // Test the output:
+  for (size_t i = 0; i < batchSize; i++) {
+    float a = TES.at({i, 0});
+    EXPECT_NEAR(resH.at({i, 0}), (std::log(a)), 0.1);
```

Comment:
Is `0.1` the right allowed error here? Given the very narrow range of the test data I have a feeling this should be much lower in order to make sure the test is actually doing something.

Reply:
changed to 0.02